### PR TITLE
JAMES-2127: Inspection guave's functions replaced by java 8

### PR DESCRIPTION
**Pull request by quynhn**
It should:
- Replace guave Optional
- No more FluentIterable.from for collection
- No Guave Supplier
- No Guave Predicate

### CODE REVIEW

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java [hunk 1]
```diff
@@ -69,6 +68,14 @@
 
 public class StoreMessageIdManager implements MessageIdManager {
     private static final Logger LOGGER = LoggerFactory.getLogger(StoreMessageIdManager.class);
+    public static <T, S> Predicate<T> compose(Predicate<S> predicate, Function<T, S> function) {
+        return input -> predicate.test(function.apply(input));
+    }
+
+    static<T> Predicate<T> not(Predicate<T> p) {
+        return t -> !p.test(t);
+    }
```
Comment (rouazana) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java]: these 2 methods should be in java8-utils, no?

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java [hunk 2]
```diff
@@ -269,23 +286,22 @@ private void save(MailboxSession mailboxSession, MessageIdMapper messageIdMapper
...
+            .findFirst();
 
         if (mailboxForbidden.isPresent()) {
             throw new MailboxNotFoundException("Mailbox with Id " + mailboxForbidden.get() + " does not belong to session");
         }
     }
 
     private Predicate<MailboxId> isMailboxOfOtherUser(MailboxSession mailboxSession, MailboxMapper mailboxMapper) {
-        return Predicates.not(mailboxBelongsToUser(mailboxSession, mailboxMapper));
+        return not(mailboxBelongsToUser(mailboxSession, mailboxMapper));
```
Comment (rouazana) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java]: in this case you can stick using Predicates, as it would allow you to not define yourself the not() method?

File: mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java [hunk 3]
```diff
@@ -117,21 +117,22 @@ public static int getInitParameterAsStrictlyPositiveInteger(String condition, in
...
-        return getInitParameterAsStrictlyPositiveInteger(condition, Optional.<String>absent());
+        return getInitParameterAsStrictlyPositiveInteger(condition, Optional.empty());
     }
 
     public static int getInitParameterAsStrictlyPositiveInteger(String condition, Optional<String> defaultValue) throws MessagingException {
-        Optional<String> value = Optional.fromNullable(condition)
-            .or(defaultValue);
+        String value = Optional.ofNullable(condition)
+            .filter(Strings::isNullOrEmpty)
+            .orElse(defaultValue.orElse(null));
```
Comment (rouazana) [inline mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java]: why this change?

File: mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java [hunk 4]
```diff
@@ -117,21 +117,22 @@ public static int getInitParameterAsStrictlyPositiveInteger(String condition, in
...
 
-        if (Strings.isNullOrEmpty(value.orNull())) {
+        if (Strings.isNullOrEmpty(value)) {
```
Comment (rouazana) [inline mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java]: this is useless now that you filter, but then you don't get the error message, so please keep the old behavior

File: server/container/util-java8/src/main/java/org/apache/james/util/OptionalConverter.java [hunk 5]
```diff
@@ -35,12 +35,12 @@
         return optional;
     }
 
-    public static <T> Optional<T> fromGuava(com.google.common.base.Optional<T> guava) {
-        return Optional.ofNullable(guava.orNull());
+    public static <T> Optional<T> fromGuava(Optional<T> guava) {
```
Comment (rouazana) [inline server/container/util-java8/src/main/java/org/apache/james/util/OptionalConverter.java]: these methods should be removed now!

File: server/container/util-java8/src/main/java/org/apache/james/util/OptionalConverter.java [hunk 6]
```diff
@@ -35,12 +35,12 @@
...
+        return Optional.ofNullable(guava.orElse(null));
     }
 
-    public static <T> com.google.common.base.Optional<T> toGuava(Optional<T> java) {
-        return com.google.common.base.Optional.fromNullable(java.orElse(null));
+    public static <T> Optional<T> toGuava(Optional<T> java) {
```
Comment (rouazana) [inline server/container/util-java8/src/main/java/org/apache/james/util/OptionalConverter.java]: here also

File: server/container/util-java8/src/test/java/org/apache/james/util/OptionalConverterTest.java [hunk 7]
```diff
@@ -92,15 +91,15 @@ public void fromGuavaShouldThrowWhenGuavaIsNull() {
 
     @Test
     public void fromGuavaShouldReturnEmptyWhenGuavaIsEmpty() {
-        Optional<String> fromGuava = OptionalConverter.fromGuava(com.google.common.base.Optional.<String> absent());
+        Optional<String> fromGuava = OptionalConverter.fromGuava(Optional.<String> empty());
```
Comment (rouazana) [inline server/container/util-java8/src/test/java/org/apache/james/util/OptionalConverterTest.java]: also these tests

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentMapper.java [hunk 8]
```diff
@@ -111,7 +109,7 @@ private Attachment attachment(Row row) {
 
         return FluentFutureStream
             .of(attachments)
-            .flatMap(OptionalConverter::toStream)
+            .flatMap(OptionalUtils::toStream)
```
Comment (rouazana) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentMapper.java]: then we don't need neither to keep OptionalConverter::toStream? Maybe we can get rid of all the OptionalConverter class?

File: server/container/util-java8/src/main/java/org/apache/james/util/OptionalUtils.java [hunk 9]
```diff
@@ -35,14 +35,6 @@
...
     }
 
-    public static <T> Optional<T> fromGuava(Optional<T> guava) {
-        return Optional.ofNullable(guava.orElse(null));
-    }
-
-    public static <T> Optional<T> toGuava(Optional<T> java) {
-        return Optional.ofNullable(java.orElse(null));
-    }
-
```
Comment (rouazana) [inline server/container/util-java8/src/main/java/org/apache/james/util/OptionalUtils.java]: oh ok that's done here :)

File: server/data/data-library/src/main/java/org/apache/james/rrt/lib/MappingsImpl.java [hunk 10]
```diff
@@ -44,6 +43,10 @@
 
     private static final long serialVersionUID = 1L;
 
+    private static<T> Predicate<T> not(Predicate<T> p) {
+        return t -> !p.test(t);
+    }
```
Comment (rouazana) [inline server/data/data-library/src/main/java/org/apache/james/rrt/lib/MappingsImpl.java]: duplicated definition

File: mailet/standard/src/main/java/org/apache/james/transport/mailets/UseHeaderRecipients.java [hunk 11]
```diff
@@ -174,15 +173,10 @@ public String getMailetInfo() {
...
-
-        for (Address address: addressList) {
-            mailboxList.addAll(convertAddressToMailboxCollection(address));
-        }
-
-        return FluentIterable.from(mailboxList.build())
-            .transform(this::toMailAddress)
-            .toList();
+        return addressList.stream()
+            .flatMap(address -> convertAddressToMailboxCollection(address).stream())
```
Comment (chibenwa) [inline mailet/standard/src/main/java/org/apache/james/transport/mailets/UseHeaderRecipients.java]: Couldn't convertAddressToMailboxCollection directly return a stream?

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/utils/MailAddressCollectionReader.java [hunk 12]
```diff
@@ -20,31 +20,32 @@
...
 
     public static Set<MailAddress> read(String condition) {
         Preconditions.checkArgument(!Strings.isNullOrEmpty(condition));
-        return FluentIterable.from(Splitter.onPattern("(,| |\t)")
-            .split(condition))
+        return Splitter.onPattern("(,| |\t)").splitToList(condition)
+            .stream()
             .filter(s -> !Strings.isNullOrEmpty(s))
-            .transform(s -> {
+            .map(s -> {
```
Comment (chibenwa) [inline mailet/standard/src/main/java/org/apache/james/transport/matchers/utils/MailAddressCollectionReader.java]: Extract the try/catch please

File: protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractAuthProcessor.java [hunk 13]
```diff
@@ -169,7 +170,7 @@ protected static AuthenticationAttempt delegation(String authorizeId, String aut
     }
 
     protected static AuthenticationAttempt noDelegation(String authenticationId, String password) {
-        return new AuthenticationAttempt(Optional.<String>absent(), authenticationId, password);
+        return new AuthenticationAttempt(Optional.<String>empty(), authenticationId, password);
```
Comment (chibenwa) [inline protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractAuthProcessor.java]: <String> still needed?

File: server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTable.java [hunk 14]
```diff
@@ -176,9 +174,9 @@ public String asKey() {
     @Override
     protected String mapAddressInternal(String user, String domain) throws RecipientRewriteTableException {
         Mappings mappings = retrieveMappings(user, domain)
-            .or(() -> retrieveMappings(WILDCARD, domain)
-                    .or(() -> retrieveMappings(user, WILDCARD)
-                            .or(MappingsImpl.empty())));
+            .orElse(retrieveMappings(WILDCARD, domain)
```
Comment (chibenwa) [inline server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTable.java]: OrElseGet please

We want to keep lazy evaluation

File: server/data/data-library/src/main/java/org/apache/james/rrt/lib/MappingsImpl.java (hunk 10 above)
Comment (chibenwa) [reply server/data/data-library/src/main/java/org/apache/james/rrt/lib/MappingsImpl.java]: I think it is better to have a "!" in a lambda rather than a not...

File: server/data/data-library/src/main/java/org/apache/james/rrt/lib/MappingsImpl.java [hunk 15]
```diff
@@ -125,7 +124,8 @@ private MappingsImpl(Collection<Mapping> mappings) {
     
     @Override
     public Iterable<String> asStrings() {
-        return FluentIterable.from(mappings).transform(Mapping::asString);
+        return mappings.stream()
+            .map(Mapping::asString)::iterator;
```
Comment (chibenwa) [inline server/data/data-library/src/main/java/org/apache/james/rrt/lib/MappingsImpl.java]: Oups?

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RecipientRewriteTableProcessor.java [hunk 16]
```diff
@@ -38,32 +42,30 @@
...
-import com.google.common.base.Function;
-import com.google.common.base.Optional;
 import com.google.common.base.Preconditions;
-import com.google.common.base.Predicate;
-import com.google.common.collect.FluentIterable;
 import com.google.common.collect.ImmutableList;
 
 public class RecipientRewriteTableProcessor {
     private static final Logger LOGGER = LoggerFactory.getLogger(RecipientRewriteTableProcessor.class);
+    private static final boolean parallel = true;
```
Comment (chibenwa) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RecipientRewriteTableProcessor.java]: parrallel = false and extract iterator -> stream

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/Resend.java [hunk 17]
```diff
@@ -367,13 +365,14 @@ public String getMessage(Mail originalMail) throws MessagingException {
...
-            return Optional.absent();
+            return Optional.empty();
         }
 
-        return FluentIterable.from(AddressExtractor.withContext(getMailetContext())
+        return AddressExtractor.withContext(getMailetContext())
                 .allowedSpecials(ImmutableList.of("postmaster", "sender", "null", "unaltered"))
-                .extract(replyTo))
-            .first();
+                .extract(replyTo)
```
Comment (chibenwa) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/Resend.java]: indent?

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remoteDelivery/AddressesArrayToMailAddressListConverter.java [hunk 18]
```diff
@@ -40,19 +39,20 @@
         if (addresses == null) {
             return ImmutableList.of();
         }
-        return FluentIterable.from(Arrays.asList(addresses))
-            .transform(address -> toMailAddress(address))
+        return Arrays.asList(addresses)
+            .stream()
+            .map(address -> toMailAddress(address))
             .filter(Optional::isPresent)
```
Comment (chibenwa) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remoteDelivery/AddressesArrayToMailAddressListConverter.java]: .flatMap(OptionalUtils::toStream) is better IMO

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remoteDelivery/DelaysAndMaxRetry.java [hunk 19]
```diff
@@ -80,10 +79,13 @@ private static DelaysAndMaxRetry addExtraAttemptToLastDelay(int intendedMaxRetri
             // Use default delayTime.
             return ImmutableList.of(new Delay());
         }
-        ImmutableList<String> delayStrings = FluentIterable.from(Splitter.on(',')
+
+        ImmutableList<String> delayStrings = Splitter.on(',')
             .omitEmptyStrings()
-            .split(delaysAsString))
-            .toList();
+            .splitToList(delaysAsString)
```
Comment (chibenwa) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remoteDelivery/DelaysAndMaxRetry.java]: The .stream().collect(...) seems unneeded.

File: server/mailet/mailets/src/main/java/org/apache/james/transport/util/MailAddressUtils.java [hunk 20]
```diff
@@ -45,17 +45,17 @@
...
     public static InternetAddress[] toInternetAddressArray(List<MailAddress> mailAddresses) {
         return iterableOfInternetAddress(mailAddresses)
-            .toArray(InternetAddress.class);
+            .toArray(InternetAddress[]::new);
     }
 
-    private static FluentIterable<InternetAddress> iterableOfInternetAddress(List<MailAddress> mailAddresses) {
-        return FluentIterable.from(mailAddresses)
-            .transform(MailAddress::toInternetAddress);
+    private static Stream<InternetAddress> iterableOfInternetAddress(List<MailAddress> mailAddresses) {
```
Comment (chibenwa) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/util/MailAddressUtils.java]: it is no more an iterable but a stream.

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesCreationProcessor.java [hunk 21]
```diff
@@ -50,13 +48,12 @@
...
 import org.apache.james.metrics.api.MetricFactory;
 import org.apache.james.metrics.api.TimeMetric;
-import org.apache.james.util.OptionalConverter;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 import com.github.fge.lambdas.Throwing;
 import com.github.fge.lambdas.functions.ThrowingFunction;
 import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
```
Comment (chibenwa) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesCreationProcessor.java]: These imports reordering looks wrong

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MessageFactory.java [hunk 22]
```diff
@@ -219,8 +219,8 @@ private Attachment fromMailboxAttachment(MessageAttachment attachment) {
                     .blobId(BlobId.of(attachment.getAttachmentId().getId()))
                     .type(attachment.getAttachment().getType())
                     .size(attachment.getAttachment().getSize())
-                    .name(attachment.getName().orNull())
-                    .cid(attachment.getCid().transform(Cid::getValue).orNull())
+                    .name(attachment.getName().orElse(null))
```
Comment (chibenwa) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MessageFactory.java]: name should accept an Optional.

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MessageFactory.java [hunk 23]
```diff
@@ -219,8 +219,8 @@ private Attachment fromMailboxAttachment(MessageAttachment attachment) {
...
+                    .cid(attachment.getCid().map(Cid::getValue).orElse(null))
```
Comment (chibenwa) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MessageFactory.java]: Cid should accept an optional

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java (hunk 2 above)
Comment (quynhn) [reply mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java]: Predicates is guava and it accept just guava's Predicate. That's why we have to define our-self method

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java (hunk 1 above)
Comment (quynhn) [reply mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java]: Removed, we can use .negate() directly

File: mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java (hunk 3 above)
Comment (quynhn) [reply mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java]: defaultValue is an Optional
The java Optional does not have default value as an Optional.
Anyway, I remove "filter" which should be confuse

File: mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java (hunk 4 above)
Comment (quynhn) [reply mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java]: No, it's not useless if defaultValue is null also

File: server/container/util-java8/src/main/java/org/apache/james/util/OptionalConverter.java (hunk 5 above)
Comment (quynhn) [reply server/container/util-java8/src/main/java/org/apache/james/util/OptionalConverter.java]: Yes, it's done by next commit

File: server/container/util-java8/src/main/java/org/apache/james/util/OptionalConverter.java (hunk 6 above)
Comment (quynhn) [reply server/container/util-java8/src/main/java/org/apache/james/util/OptionalConverter.java]: idem

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentMapper.java (hunk 8 above)
Comment (quynhn) [reply mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentMapper.java]: Basically yes, we can but later on we have to detail on using like
```
.flatMap(OptionalUtils::toStream)
```
should be

```
.flatMap(op -> op.map(Stream::of).orElse(Stream.of()))
```

It makes to code more complicated, so I keeped the old one

File: server/data/data-library/src/main/java/org/apache/james/rrt/lib/MappingsImpl.java (hunk 10 above)
Comment (quynhn) [reply server/data/data-library/src/main/java/org/apache/james/rrt/lib/MappingsImpl.java]: Deleted such as we can use .negative() from java 8

**Comment by quynhn** [comment]
>can you check if can now remove some guava dependencies?
Yes, I will scan them all but on next Monday

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentMapper.java (hunk 8 above)
Comment (chibenwa) [reply mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentMapper.java]: I like the way it was renamed OptionalUtils.

**Comment by chibenwa** [comment]
:-( Please avoid force push on big current PR as much as you can....

I may need to read it again from the beginning, and it took me more than 30 minutes 😠

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/MessageAttachmentRepresentation.java [hunk 24]
```diff
@@ -40,9 +39,9 @@ public static Builder builder() {
     public static MessageAttachmentRepresentation fromAttachment(MessageAttachment attachment) {
         return builder()
             .attachmentId(attachment.getAttachmentId())
-            .cid(OptionalConverter.fromGuava(attachment.getCid()))
+            .cid(attachment.getCid())
             .isInline(attachment.isInline())
-            .name(attachment.getName().orNull())
+            .name(attachment.getName().orElse(null))
```
[the remark this answers is not in the document]
Comment (quynhn) [reply mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/MessageAttachmentRepresentation.java]: I removed that function such as it's no longer in use anymore so we don't need to support a method for another unused method IMO

File: mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAAnnotationMapper.java [hunk 25]
```diff
@@ -101,30 +103,21 @@ public JPAAnnotationMapper(EntityManagerFactory entityManagerFactory) {
...
+            return keys.stream()
+                .flatMap(key -> getEntityManager()
+                    .createNamedQuery("retrieveByKeyLike", JPAMailboxAnnotation.class)
+                    .setParameter("idParam", jpaId.getRawId())
+                    .setParameter("keyParam", key.asString() + '%')
+                    .getResultList()
+                    .stream()
+                    .map(READ_ROW)
+                    .filter(predicateFunction.apply(key)))
+                .collect(Guavate.toImmutableList());
```
[the remark this answers is not in the document]
Comment (quynhn) [reply mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAAnnotationMapper.java]: No we can't because (l 114) using key which is the parameter of flatMap

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java [hunk 26]
```diff
@@ -57,18 +59,22 @@
...
 import com.google.common.collect.Sets;
 import com.google.common.collect.Sets.SetView;
 
 public class StoreMessageIdManager implements MessageIdManager {
     private static final Logger LOGGER = LoggerFactory.getLogger(StoreMessageIdManager.class);
+    public static <T, S> Predicate<T> compose(Predicate<S> predicate, Function<T, S> function) {
+        return input -> predicate.test(function.apply(input));
+    }
+
+    static<T> Predicate<T> not(Predicate<T> p) {
```
[the remark this answers is not in the document]
Comment (quynhn) [reply mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java]: It's removed and using negate()

File: mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java [hunk 27]
```diff
@@ -117,21 +117,22 @@ public static int getInitParameterAsStrictlyPositiveInteger(String condition, in
...
 
     public static int getInitParameterAsStrictlyPositiveInteger(String condition) throws MessagingException {
-        return getInitParameterAsStrictlyPositiveInteger(condition, Optional.<String>absent());
+        return getInitParameterAsStrictlyPositiveInteger(condition, Optional.empty());
     }
 
     public static int getInitParameterAsStrictlyPositiveInteger(String condition, Optional<String> defaultValue) throws MessagingException {
-        Optional<String> value = Optional.fromNullable(condition)
-            .or(defaultValue);
+        String value = Optional.ofNullable(condition)
```
[the remark this answers is not in the document]
Comment (quynhn) [reply mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java]: I tried to keep but as I answer:
>defaultValue is an Optional
>The java Optional does not have default value as an Optional.
So I changed its type

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remoteDelivery/AddressesArrayToMailAddressListConverter.java (hunk 18 above)
Comment (quynhn) [reply server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remoteDelivery/AddressesArrayToMailAddressListConverter.java]: And added one more test

**Comment by quynhn** [comment]
>Yes, I will scan them all but on next Monday

In fact, we use guava for Precondition, MoreObjects and Objects ... which are very common and popular so I cannot remove guava dependency from any module

**Comment by quynhn** [comment]
>:-( Please avoid force push on big current PR as much as you can....
>
>I may need to read it again from the beginning, and it took me more than 30 minutes :angry:

Yes

**Comment by quynhn** [comment]
I've just had to push force such as a failure building due to conflict with master

**Comment by chibenwa** [comment]
```

Failed tests: 
  GetAnnotationProcessorTest.processShouldGetAnnotationsAndReturnCompleteResponseWithTheLongestEntryInfoWhenLimitMaxsize:203 
Expecting:
  <[METADATA LONGENTRIES]>
to contain only:
  <[METADATA LONGENTRIES]>
elements not found:
  <[METADATA LONGENTRIES]>
and elements not expected:
  <[METADATA LONGENTRIES]>


Tests run: 508, Failures: 1, Errors: 0, Skipped: 5
```

Could you have a look @quynhn ? It might be related with your code...

**Comment by quynhn** [comment]
Have to rebase on master due to compilation error after merging with master

**Comment by chibenwa** [comment]
```
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   ParsingWritingObjectsTest.parsingJsonShouldWorkOnSubMessage:57 » JsonMapping C...
[INFO] 
[ERROR] Tests run: 742, Failures: 0, Errors: 1, Skipped: 2

[INFO] Apache James :: Server :: JMAP ..................... FAILURE [01:13 min]
```

**Comment by quynhn** [comment]
test this please

**Comment by chibenwa** [comment]
Merged
